src/main/java/io/github/mishkun/ideavimsneak/IdeaVimSneakExtension.kt

```diff
@@ -135,9 +145,18 @@ class IdeaVimSneakExtension : VimExtension {
         },
         BACKWARD(-1) {
             override fun findBiChar(charSequence: CharSequence, position: Int, charone: Char, chartwo: Char): Int? {
+                var match: Boolean
+
                 for (i in (position + offset) downTo 0) {
-                    if (charSequence[i] == charone &&
-                            charSequence[i + 1] == chartwo) {
+                    match = charSequence[i].equals(charone, ignoreCase = OptionsManager.ignorecase.isSet) &&
+                            charSequence[i + 1].equals(chartwo, ignoreCase = OptionsManager.ignorecase.isSet)
+                    if (OptionsManager.ignorecase.isSet && OptionsManager.smartcase.isSet) {
```

Comment:
If you want to try be more idiomatic, you could use `when` statement instead of nested if's

Reply:
I looked into it and I'm not too sure how to go about it.
What I tried was less readable than the nested if but I might be missing something.

Reply:
No problem, code looks good enough once extracted into function 👍